src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -58,30 +67,50 @@ class MetadataBasedItemList extends React.Component<Props, State> {
             }
 
             if (index === 1) {
-                return FILENAME_COLUMN_WIDTH;
+                return FILE_NAME_COLUMN_WIDTH;
             }
 
-            const availableWidth = width - FILENAME_COLUMN_WIDTH - FILE_ICON_COLUMN_WIDTH; // total width minus width of sticky columns
+            const availableWidth = width - FILE_NAME_COLUMN_WIDTH - FILE_ICON_COLUMN_WIDTH; // total width minus width of sticky columns
             // Maintain min column width, else occupy the rest of the space equally
             return Math.max(availableWidth / metadataColumnsToShow.length, MIN_METADATA_COLUMN_WIDTH);
         };
     }
 
-    getGridCellData(columnIndex: number, rowIndex: number): Element<typeof FileIcon> | string {
-        const { currentCollection, metadataColumnsToShow }: Props = this.props;
-        const { items } = currentCollection;
-        const { name = '', metadata = {} } = items[rowIndex - 1];
+    handleOnClick(item: FlattenedMetadataQueryResponseEntry): void {
+        const { onItemClick }: Props = this.props;
+        /*
+            - @TODO: Remove permissions object once its part of API response.
+            - In Content Explorer element, if can_preview permission is false, there is no action taken onClick(item).
+            - Until the response has permissions, add "can_preview: true" so that users can click to launch the Preview modal. If users don't have access, they will see the error when Preview loads.
+        */
+        const permissions = { can_preview: true };
+        const itemWithPreviewPermission = { ...item, ...{ permissions } };
```

Comment:
Would it work if you just did `{ ...item, permissions }`?

Reply:
Good catch! Should work.